Dockerfile

```diff
@@ -0,0 +1,43 @@
+FROM python:3.10.12-slim-bullseye AS local
+
+# pin exact version so docker invalidates on every change of this file
+RUN pip3 install pip==23.2.1
+
+ENV PIP_DISABLE_PIP_VERSION_CHECK=1 \
+    PYTHONDONTWRITEBYTECODE=1
+
+COPY requirements/dev-3.10.txt /app/requirements/dev-3.10.txt
+RUN set -e; \
+        buildDeps=' \
+                # n/a
+        '; \
+        runtimeDeps=' \
+                # TODO cleanup: trying to figure out container \
+                protobuf-compiler \
+                build-essential \
```

Comment:
this should be removed before merging. i added it for testing within the container. adding make targets to the repo that can utilize docker/compose would eliminate the need for this, but i haven't gotten there yet.